### PR TITLE
Brocade/Foundry ADX/ServerIron support

### PR DESCRIPTION
**Pull request by xbeaudouin**
Add some counter to MIB support the number of connection to a load balancer

This allow all ServerIron XL and/or ADX (from Brocade) to graph CPU and Memory.
This allow as well to graph number connection the LB has done.

### CODE REVIEW

**Comment by ekoyle** [comment]
It looks like you are working on several separate things in this PR...
 1) treat brocade serveriron devices as ironware (should work for basic polling, not load balancer specific stuff)
 2) treat nos (Brocade VDX) devices the same as ironware for purposes of processor polling
 3) adding support for pktj/packet journey
 4) modified  includes/snmp.inc.php to treat 32/64-bit counters alike (I'm not sure this is a safe change, I think the 32-bit counters were intentionally ignored at some point)

This doesn't look like it will match on ADXes, maybe you should match on 'ServerIron' instead?

SNMPv2-MIB::sysDescr.0 = STRING: Brocade Communications Systems, Inc. ServerIron ADX 1008-1, TrafficWork Version 12.5.01gT401

**Comment by vizay** [comment]
As for the OS polling, isn't it cleaner to keep polling of ADX-hardware separate from VDX since it's different devices?

I know the vdx-stuff is named nos which might be a bit of a bummer, but it might easily get hard to follow what's happening if that file suddenly is responsible for OS polling of two different kinds of devices.

Just a personal opinion :)

**Comment by f0o** [comment]
Each OS needs their own file. Mixing it causes confusion and makes debugging a lot harder.
Tagging @librenms/reviewers

File: includes/discovery/mempools/ironware-dyn.inc.php [hunk 1]
```diff
@@ -1,6 +1,6 @@
 <?php
 
-if ($device['os'] == 'ironware' || $device['os_type'] == 'ironware') {
+if ($device['os'] == 'ironware' || $device['os_type'] == 'ironware' || $device['os'] == 'nos') {
```
Comment (vizay) [inline includes/discovery/mempools/ironware-dyn.inc.php]: Seeing this now makes me realize the last part of the IF-statement stands in conflict with nos.inc.php that's also does discovery for nos.

Consider a Brocade VDX device, will it now be discovered as ironware or nos?

**Comment by xbeaudouin** [comment]
Well I made a mistake with the "clickostuff" and I wanted to make 2 differents pull request but for an unknown reason (sorry for my low understanding of git and github) github merged all of them into one.

I can redo this pull request from scratch if you like (I usualy better work with unified patch than this unfortunatly).

We have 2 kind of Server Iron (ADX or XL) that have exactly the same MIB as all Foundry/Brocade stuff : 

SNMPv2-MIB::sysDescr.0 = STRING: Brocade Communications Systems, Inc. ServerIron ADX 1216-4-PREM, TrafficWork Version 12.5.01cT403

SNMPv2-MIB::sysDescr.0 = STRING: Foundry Networks, Inc. ServerIron Switch, Iron Software Version 07.5.00T12 Compiled on Mar 17 2005 at 12:08:20 labeled as SLB07500

Either we match on 'ServerIron' only and we get this working or doing it another way.

I am opened to your suggestions.

**Comment by vizay** [comment]
I see your predicament now, sorry for not noticing quicker!

We should look into patching it so neither VDX switches, nor anything else matches only on "Brocade Communications Systems" :)

I can test a better matching on my side and propose a code change that you can submit, in case you don't have any VDX-switches to try it on yourself :)

**Comment by xbeaudouin** [comment]
@vizay I don't have any VDX switches on my side. Those ServerIron are the only brocade/foundry stuff we have.
Also I wanted to avoid to duplicate the code :)

**Comment by ekoyle** [comment]
@xbeaudouin
<https://github.com/librenms/librenms/issues?q=is%3Apr+is%3Aopen+author%3Axbeaudouin>
If
you make any changes to your branch, it will update your pull request.  The
basics of creating a new branch are:

git checkout -b <new_branch_name>
#<make changes and commit>
git push your_remote new_branch_name
# go back to your normal branch
git checkout master

Does anyone remember what devices needed the additional matching for the
VDX devices (nos module)?  I don't have that many, but all of mine say
'Brocade VDX' in the sysdescr.

For reference, here are is a selection of sysdescr from brocade devices on
our network:

VDX:
SNMPv2-MIB::sysDescr.0 = STRING: Brocade VDX Switch, BR-VDX6720-24, Network
Operating System Software Version 4.1.3b.
SNMPv2-MIB::sysDescr.0 = STRING: Brocade VDX Switch, BR-VDX6720-60, Network
Operating System Software Version 4.1.3b.

NetIron devices:
SNMPv2-MIB::sysDescr.0 = STRING: Brocade MLXe (System Mode: MLX), IronWare
Version V5.6.0fT163 Compiled on Jul  7 2015 at 23:40:24 labeled as V5.6.00ff
SNMPv2-MIB::sysDescr.0 = STRING: Brocade NetIron MLX (System Mode: MLX),
IronWare Version V5.6.0fT163 Compiled on Jul  7 2015 at 23:40:24 labeled as
V5.6.00ff
SNMPv2-MIB::sysDescr.0 = STRING: Brocade NetIron CER, IronWare Version
V5.6.0fT183 Compiled on Jul 07 2015 at 23:55:53 labeled as V5.6.00ff

FastIron devices:
SNMPv2-MIB::sysDescr.0 = STRING: Brocade Communications Systems, Inc.
ICX6610-24, IronWare Version 08.0.10jT7f1 Compiled on Mar 26 2015 at
21:39:28 labeled as FCXS08010j
SNMPv2-MIB::sysDescr.0 = STRING: Brocade Communications Systems, Inc.
ICX6430-48-HPOE, IronWare Version 08.0.10jT311 Compiled on Mar 26 2015 at
21:09:56 labeled as ICX64S08010j
SNMPv2-MIB::sysDescr.0 = STRING: Brocade Communications Systems, Inc.
TurboIron-X24, IronWare Version 07.4.00eT201 Compiled on Dec 11 2013 at
19:28:34 labeled as TIS07400e
SNMPv2-MIB::sysDescr.0 = STRING: Brocade Communications Systems, Inc.
ICX7750-48F, IronWare Version 08.0.30bT203 Compiled on Jul 31 2015 at
02:05:32 labeled as SWR08030b
SNMPv2-MIB::sysDescr.0 = STRING: Foundry Networks, Inc. FES4802-ILP,
IronWare Version 04.1.00Tc1 Compiled on Aug 10 2008 at 11:29:59 labeled as
FEL04100
SNMPv2-MIB::sysDescr.0 = STRING: Brocade Communications Systems, Inc.
FWS624-POE, IronWare Version 07.4.00fT7e1 Compiled on Mar 26 2014 at
06:51:54 labeled as FWS07400f

ServerIron devices (load balancers):
SNMPv2-MIB::sysDescr.0 = STRING: Brocade Communications Systems, Inc.
ServerIron ADX 1008-1, TrafficWork Version 12.5.01gT401


On Fri, Jan 8, 2016 at 3:00 AM, Xavier Beaudouin <notifications@github.com>
wrote:

> @vizay <https://github.com/vizay> I don't have any VDX switches on my
> side. Those ServerIron are the only brocade/foundry stuff we have.
> Also I wanted to avoid to duplicate the code :)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/librenms/librenms/pull/2712#issuecomment-169951685>.
>



-- 
Eldon Koyle

**Comment by vizay** [comment]
we have a variety of VDX switches and I can confirm they all report "Brocade VDX" and nothing else.

Your pull request #2720 hence makes a lot of sense, not only making stuff cleaner but also making sure we don't accidently match something else like a VDX :)

**Comment by paulgear** [comment]
Note also that this needs updating to use register_mibs instead of poll_mibs as of MIB-based poller alpha 2 (merged last month).

**Comment by laf** [comment]
@xbeaudouin any update?

**Comment by f0o** [comment]
@xbeaudouin bump

**Comment by xbeaudouin** [comment]
Hi.

On holidays (new twin born) will see and try to port this into master.

Please hold on.

**Comment by f0o** [comment]
@xbeaudouin congratulations mate !!

**Comment by vizay** [comment]
@xbeaudouin Congrats!

Now to the matter at hand, this PR get's more and more confusing. Have you dropped the ADX stuff completely?

You're still doing stuff in includes/polling/os/nos.inc.php which doesn't make sense.

Have you accidentally done work on the wrong branch perhaps? :)

**Comment by xbeaudouin** [comment]
Well I make a mistake on this. I think it is better to start on a clean work. So I will close this pull request and make a new one :)
